### DISCUSSION TITLE
Saving page using the extracted dictionary

### DISCUSSION THREAD

**Discussion by SrijithBalachander** [pymupdf/PyMuPDF#1899] [Looking for help] [answered]
Hi, I just started exploring PyMuPDF for my use case and have been struggling to do one simple thing. I extracted a page of a pdf as a dictionary using `get_text("dict")` where I perform a few font-size changes. I would like to save these changes back into the page or create a new page and enter text using the dictionary. How can I go about doing this?
Thank you!

**Comment by JorjMcKie** [comment, marked as answer]
Iterate over the dictionary - there are 3 hierarchy levels: blocks, lines and spans.
For every span perform a `page.insert_text()`, taking the following argument values:
* insertion point: span["origin"]
* text: span["text"]
* fontsize: span["size"]
* fontname is a trickier thing:
    - if you can use one of the Base-14 fonts, Times-Roman, Helvetica, Courier. Watch for choosing the right font weight / style by inspecting the span["flags"] value.
    - re-using existing fonts is difficult if writing to a new file - save this for a time when you have more experience.
    - using file-based new fonts is also not straightforward: you must first insert the font to the page, assign a fontname to it, e.g. "F1", then use fontname="F1" in the insert_text instruction.

Please also note that there is a font replacement utility [here](https://github.com/pymupdf/PyMuPDF-Utilities/tree/master/font-replacement) which does a very similar thing. Maybe you can draught insight from it.
